### PR TITLE
Alloc fixes

### PR DESCRIPTION
**Pull request by jforissier**

### CODE REVIEW

**Comment by jforissier** [comment]
Fixed commit message of first patch.

File: lib/libutils/isoc/bget_malloc.c [hunk 1]
```diff
@@ -312,6 +312,7 @@ static void *raw_malloc(size_t hdr_size, size_t ftr_size, size_t pl_size,
 			struct bpoolset *poolset)
 {
 	void *ptr = NULL;
+	bufsize bs;
 	size_t s;
```
Comment (jenswi-linaro) [inline lib/libutils/isoc/bget_malloc.c]: Wouldn't it be enough to change the type of `s` to `bufsize` instead?

Comment (jforissier) [reply lib/libutils/isoc/bget_malloc.c]: Indeed. I thought the `_OVERFLOW` macros had to operate on arguments of the same type, but apparently not! Will fix.

**Comment by jforissier** [comment]
Updated.
I think it doesn't really make sense to keep two patches, how about I squash them together? (I would still explain the two failure cases I met with xtest 8033 on the one hand and 8034/8042 on the other hand).

File: lib/libutils/isoc/bget_malloc.c (hunk 1 above)
Comment (jenswi-linaro) [reply lib/libutils/isoc/bget_malloc.c]: Yeah, it took some extra effort to support that. :-)

**Comment by jenswi-linaro** [comment]
Please squash the patches.
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by jforissier** [comment]
Travis fails but I can''t reproduce the crash in my environment :( I will need to add traces.

**Comment by jforissier** [comment]
Looks like the `ADD_OVERFLOW` macro doesn't work with GCC 4.9.
```C
        DMSG("hdr_size=%zu ftr_size=%zu pl_size=%zu", hdr_size, ftr_size,
             pl_size);
        if (ADD_OVERFLOW(pl_size, hdr_size, &s))
                goto out;
```
```
D/TC:0 raw_realloc:390 hdr_size=0 ftr_size=0 pl_size=8
E/TC:0 assertion 'p' failed at lib/libutils/isoc/bget_malloc.c:935 <malloc_add_pool>
E/TC:0 Panic at core/kernel/assert.c:51 <_assert_break>
E/TC:0 Call stack:
E/TC:0  0x0e10b235 print_kernel_stack at optee_os/core/arch/arm/kernel/unwind_arm32.c:377
E/TC:0  0x0e125821 __do_panic at optee_os/core/kernel/panic.c:56 (discriminator 1)
E/TC:0  0x0e1235c7 _assert_break at :?
E/TC:0  0x0e14abf3 malloc_add_pool at optee_os/lib/libutils/isoc/bget_malloc.c:936
E/TC:0  0x0e109c09 init_runtime at optee_os/core/arch/arm/kernel/generic_boot.c:427
E/TC:0  0x0e10a991 init_primary_helper at optee_os/core/arch/arm/kernel/generic_boot.c:870
E/TC:0  0x0e10aaad generic_boot_init_primary at optee_os/core/arch/arm/kernel/generic_boot.c:932
```

**Comment by jforissier** [comment]
https://github.com/OP-TEE/optee_os/pull/2303

**Pull request merged**